.github/workflows/database-upgrade.yml

```diff
@@ -0,0 +1,44 @@
+name: Database Upgrade Test
+
+on:
+  push:
+    branches:
+      - main
+      - '[0-9]+.[0-9]+'
+  pull_request:
+    branches:
+      - main
+      - '[0-9]+.[0-9]+'
+
+jobs:
+  upgrade_test:
+    runs-on: ubuntu-latest
+    container:
+      image: domjudge/gitlabci:24.04
+    services:
+      sqlserver:
+        image: mariadb
+        ports:
+          - 3306:3306
+        env:
+          MYSQL_ROOT_PASSWORD: root
+          MYSQL_USER: domjudge
+          MYSQL_PASSWORD: domjudge
+        options: --health-cmd="healthcheck.sh --connect --innodb_initialized" --health-interval=10s --health-timeout=5s --health-retries=3
+    steps:
+      - uses: actions/checkout@v4
+      - name: Wait and Import Database
+        run: |
+          apt-get update
+          apt-get install -y default-mysql-client
+          while ! mysqladmin ping -h"sqlserver" -u"root" -p"root" --silent; do
```

Comment:
I think `GHA` will always first boot the service containers before booting the container, so is this step actually needed?